docs/learn/agents.md

```diff
@@ -14,7 +14,7 @@ In this section we will implement a toy `Librarian` agent to demonstrate key con
         If you haven't already, we recommend first reading the section on [Calls](./calls.md)
     </div>
 
-Since an agent needs to operate across multiple LLM API calls, the first concept to cover is state. You can introduce state in many ways ranging from local variables to a database.
+Since an agent needs to operate across multiple LLM API calls, the first concept to cover is state. You can introduce state in many ways ranging from local variables to a database, or short-term memory and long-term memory respectively.
```

Comment:
Hmm, it's not necessarily memory. Could just be state for the agent to work with, and I think "memory" is a bit overloaded as a term?

Curious the value add here of this additional description rather than keeping it more general to storage?

Reply:
its a common phrase used in agents, so i figured state or stateless is not quite the same as what people think when it comes to llm agents. what situation would you add state and consider it not memory?

Reply:
Hmm, I guess it's less about "memory" and more about short term vs. long term.

You could use a database for persistence that you load into local state for "working memory" for example

Reply:
What about something like:

The goal of providing state to the agent is to give it memory. For example, we can think of local variables as "working memory" and a database as "long-term memory"